### PR TITLE
Build and push operator and bundle images in CI

### PR DESCRIPTION
**Pull request by ruicao93**
- Test building images for pull request.
- Update images every 12 hours.

Signed-off-by: rcao <rcao@vmware.com>

### CODE REVIEW

File: .github/workflows/update_image.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+name: Update the antrea-operator and bundle images every 12 hours
```
Comment (salv-orlando) [inline .github/workflows/update_image.yml:1]: Do you think the periodic job will bring additional benefits?
I was thinking that perhaps updating the 'latest' tag everytime a PR is merged could be enough.

File: config/manager/kustomization.yaml [hunk 2]
```diff
@@ -4,8 +4,8 @@ apiVersion: kustomize.config.k8s.io/v1beta1
 kind: Kustomization
 images:
 - name: antrea-operator
-  newName: antrea-operator
-  newTag: latest
+  newName: antrea/antrea-operator
+  newTag: 0.0.1
 - name: caorui/antrea-operator
```
Comment (salv-orlando) [inline config/manager/kustomization.yaml:9]: nit: at some point your dockerhub's entry might be removed. Not sure if you still need it.

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -12,3 +15,14 @@ jobs:
...
+      - name: Build antrea-operator-bundle Docker image
+        run: make bundle-build
+      - name: Push images to registry
+        if: ${{ github.repository == 'vmware/antrea-operator-for-kubernetes' && github.event_name == 'push' && github.ref == 'refs/heads/main' }}
+        env:
+          DOCKER_USERNAME: ${{ secrets.DOCKER_USERNAME }}
+          DOCKER_PASSWORD: ${{ secrets.DOCKER_PASSWORD }}
+        run: |
+          echo "$DOCKER_PASSWORD" | docker login -u "$DOCKER_USERNAME" --password-stdin
+          docker push aantrea/antrea-operator:latest
```
Comment (salv-orlando) [inline .github/workflows/build.yml:27]: There is an extra 'a' in antrea

File: versioning.mk [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        DOCKER_IMG_VERSION := $(VERSION_PREFIX)$(VERSION)-$(VERSION_SUFFIX)
+else
+        DOCKER_IMG_VERSION := $(VERSION)
+endif
+
+# Default bundle image tag
+BUNDLE_IMG ?= antrea/antrea-operator-bundle:$(DOCKER_IMG_VERSION)
+
+# Image URL to use all building/pushing image targets
+IMG ?= antrea/antrea-operator:$(DOCKER_IMG_VERSION)
```
Comment (salv-orlando) [inline versioning.mk:42]: nit: can we call this variable OPERATOR_IMG?

**Review by salv-orlando — COMMENTED** [review]
LGTM. Minor comments inline.

File: config/manager/kustomization.yaml (hunk 2 above)
Comment (ruicao93) [reply config/manager/kustomization.yaml]: Rename caorui/antrea-operator to antrea/antrea-operator.

File: .github/workflows/build.yml (hunk 3 above)
Comment (ruicao93) [reply .github/workflows/build.yml]: Done.

File: versioning.mk (hunk 4 above)
Comment (ruicao93) [reply versioning.mk]: Done.

File: .github/workflows/update_image.yml (hunk 1 above)
Comment (ruicao93) [reply .github/workflows/update_image.yml]: Thanks Salvatore, I think we could remove it as you said.

**Comment by salv-orlando** [comment]
Thanks for addressing my comments!

**Review by salv-orlando — APPROVED** [review]

**Pull request merged**